File: docs/guides/examples/py/src/access_control_create_token.py
from reduct import Client, Permissions

# Create a client with the base URL and API token
client = Client("http://localhost:8383", api_token="my-token")


async def create_token():
    # Create a token with read/write access to the bucket "example-bucket"
    permissions = Permissions(
        full_access=False,
        read=["example-bucket"],
        write=["example-bucket"],
    )
    token = await client.create_token("example-bucket", permissions)
    print(f"generated token: {token}")


if __name__ == "__main__":
    import asyncio

    asyncio.run(create_token())

File: versioned_docs/version-1.16.x/examples/go/src/access_control_create_token.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	model "github.com/reductstore/reduct-go/model"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Create a token with read/write access to the bucket "example-bucket"
	permissions := model.TokenPermissions{
		FullAccess: false,
		Read:       []string{"example-bucket"},
		Write:      []string{"example-bucket"},
	}

	token, err := client.CreateToken(context.Background(), "new-token", permissions)
	if err != nil {
		panic(err)
	}

	// Print the generated token
	println("generated token:", token)
}
